src/base/memoryManagement.h

```diff
@@ -74,30 +74,34 @@ class MemoryManagement {
         /// a pointer as an argument; hence the strange syntax.
         void alloc(size_t size) {
             if (size > 0) {
+#ifndef USE_CPU_ONLY
                 if (onDevice) {
+
                     gpuError_t gpuErr = gpuMalloc((void **) &_rawPointer, size);
                     if (gpuErr != gpuSuccess) {
                         MemoryManagement::memorySummary(false, false, true, true,false);
                         std::stringstream err_msg;
                         err_msg << "_rawPointer: Failed to allocate (additional) " << size/1000000000. << " GB of memory on device";
                         GpuError(err_msg.str().c_str(), gpuErr);
                     }
-                } else {
-#ifndef CPUONLY
-                    gpuError_t gpuErr = gpuMallocHost((void **) &_rawPointer, size);
-                    if (gpuErr != gpuSuccess) {
+                } else
+#endif
+                {
+#if defined(CPUONLY) || defined(USE_CPU_ONLY)
```

Comment:
hmm, we have `CPUONLY` and `USE_CPU_ONLY`. That might be confusing. Do we need both? What was the difference again?

Reply:
I agree the naming should be changed.

USE_CPU_ONLY comes from the cmake parameter "-DBACKEND=cpu" and means: compile (the whole code) without a GPU backend. The name is like this because for the other cases it also says "USE_*".
CPUONLY is a parameter I introduced for CheckConf and CheckRand. It means: you can compile this executable with a GPU backend, but the executable will run (on cpu only) even if no GPU is present.

Reply:
I propose this:
USE_CPU_ONLY --> BACKEND_CPU
USE_HIP_NVIDIA --> BACKEND_HIP_NVIDIA
USE_HIP_AMD --> BACKEND_HIP_AMD
USE_CUDA --> BACKEND_CUDA